### ISSUE TITLE
System.AggregateException in Xamarin.iOS project

### ISSUE THREAD

**Issue by haknode** [Yortw/RSSDP#46]
I am using your library in an Xamarin.iOS project and got the following exception. The exception only occured once until now and i can not reproduce it, so i think its a minor problem: 
_'System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. ---> System.Net.Sockets.SocketException: The socket is not connected_

The exception was thrown from the Rssdp.Infrastructure.SsdpCommunicationsServer.ListenToSocket() method.

I think a ContinueWith() could solve the problem:
```
var t = TaskEx.Run(async () =>	{ ... });
t.ContinueWith(t2 =>
{
    //ignore exceptions
}, TaskContinuationOptions.OnlyOnFaulted);
```

**Comment by Yortw** [comment]
Hi,

I have seen this issue and care about it but have not had time to look at it. I will as soon as I can. Thanks for reporting it.

**Comment by Yortw** [comment]
I'm not able to reproduce this either, but I have added error handling to detect the socket closed condition and attempt to re-open the socket when that is detected. I believe your suggested fix may have prevented a crash but simply left the system in a state where it didn't respond to requests for no known reason.

While not able to reproduce the issue in production I did unit test the re-open logic, so assuming it is activated under the appropriate conditions it should work.

New build should be on Nuget shortly.
